lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -662,34 +676,19 @@ public TopFieldDocs reduce(Collection<TopFieldCollector> collectors) throws IOEx
       }
       query = rewrite(query);
       final Weight weight = createWeight(query, scoreMode, 1);
-      final List<Future<C>> topDocsFutures = new ArrayList<>(leafSlices.length);
-      for (int i = 0; i < leafSlices.length - 1; ++i) {
+      final List<FutureTask> listTasks = new ArrayList<>();
```

Comment:
Let's avoid introducing warnings about generics, FutureTask needs to be parameterized?